File: apps/node/src/fizzbuzz/application/fizz-buzz-value-command.ts
import { FizzBuzzType } from "../domain/type/fizz-buzz-type.js";
import { FizzBuzzValue } from "../domain/model/fizz-buzz-value.js";
import { FizzBuzzCommand } from "./fizz-buzz-command.js";

export class FizzBuzzValueCommand implements FizzBuzzCommand {
  private readonly _type: FizzBuzzType;
  private readonly _number: number;

  constructor(type: FizzBuzzType, number: number) {
    this._type = type;
    this._number = number;
  }

  execute(): FizzBuzzValue {
    return this._type.generate(this._number);
  }
}

File: apps/dotnet/FizzBuzz/Application/FizzBuzzValueCommand.cs
namespace FizzBuzz.Application;

using FizzBuzz.Domain.Model;
using FizzBuzz.Domain.Type;

public class FizzBuzzValueCommand : IFizzBuzzCommand
{
    private readonly IFizzBuzzType _type;

    public FizzBuzzValueCommand(IFizzBuzzType type)
    {
        _type = type;
    }

    public FizzBuzzValue ExecuteValue(int number)
    {
        return _type.Generate(number);
    }

    public FizzBuzzList ExecuteList(int count)
    {
        throw new NotSupportedException("FizzBuzzValueCommand does not support list execution.");
    }
}
